### PR TITLE
[CORE-567] add integration tests for vest module gov msgs

### PR DESCRIPTION
**Pull request by tqin7**
### Summary by CodeRabbit

**New Feature:**
- Added a new field `expectCheckTxFails` to various test functions in the `gov` package. This allows for more granular control over testing scenarios, specifically enabling tests where transaction checks are expected to fail.

**Test:**
- Enhanced the `SubmitAndTallyProposal` function in the `testutil/app/gov.go` file to handle cases where transaction checks or proposal submissions are expected to fail.
- Introduced two new test functions `TestSetVestEntry` and `TestDeleteVestEntry` in the `gov/vest_test.go` file to cover scenarios related to vest entry management.

These changes improve the robustness of our testing suite by allowing us to simulate and handle failure scenarios more effectively.

### CODE REVIEW

File: protocol/testing/e2e/gov/vest_test.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+func TestSetVestEntry(t *testing.T) {
+	tests := map[string]struct {
+		msg                      *vesttypes.MsgSetVestEntry
+		updateExistingVestEntry  bool // whether above msg should update an existing vest entry.
+		expectCheckTxFails       bool
+		expectSubmitProposalFail bool
+		expectedProposalStatus   govtypesv1.ProposalStatus
+	}{
+		"Success: create a new vest entry": {
```
Comment (ttl33) [inline protocol/testing/e2e/gov/vest_test.go:26]: do we have a test case with multiple vest entries?

thinking of cases like:
1. multiple vest entries, successfully updates some (i.e. like 1 or 2) entry
2. multiple vest entries, successfully updates all entries

File: protocol/testing/e2e/gov/vest_test.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+func TestDeleteVestEntry(t *testing.T) {
+	tests := map[string]struct {
+		msg                      *vesttypes.MsgDeleteVestEntry
+		vestEntryExists          bool // whether vest entry in above msg exists.
+		expectSubmitProposalFail bool
+		expectedProposalStatus   govtypesv1.ProposalStatus
+	}{
+		"Success": {
+			msg: &vesttypes.MsgDeleteVestEntry{
```
Comment (ttl33) [inline protocol/testing/e2e/gov/vest_test.go:174]: similar to above, 
1. define + use constants
2. write multiple vest entry deletion if possible

File: protocol/testing/e2e/gov/vest_test.go (hunk 1 above)
Comment (ttl33) [reply protocol/testing/e2e/gov/vest_test.go:26]: oh also 
3. update the same entry multiple times (ie gov msg contains multiple message for the same vest entry)
4. add and then update the same entry (i.e. no entry exists, the gov msg contains 2 msgs, the first msg inserts, then the second msg updates)

File: protocol/testing/e2e/gov/vest_test.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+					EndTime:         time.Date(2023, 10, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			expectCheckTxFails: true,
+		},
+		"Failure: invalid authority": {
+			msg: &vesttypes.MsgSetVestEntry{
+				Authority: constants.BobAccAddress.String(),
+				Entry: vesttypes.VestEntry{
+					VesterAccount:   vesttypes.ModuleName,
```
Comment (teddyding) [inline protocol/testing/e2e/gov/vest_test.go:96]: Nit: `vesttypes.ModuleName` is never used as a vester account so it's bit misleading how it's used here (See [rewartypes.VesterAccountName](https://github.com/dydxprotocol/v4-chain/blob/604ba626f00be9b7a145b702fee3e2dea42727ce/protocol/x/rewards/types/keys.go#L33C15-L33C15) or [CommunityVesterAccountName](https://github.com/dydxprotocol/v4-chain/blob/604ba626f00be9b7a145b702fee3e2dea42727ce/protocol/x/vest/types/keys.go#L30))

File: protocol/testing/e2e/gov/vest_test.go (hunk 1 above)
Comment (tqin7) [reply protocol/testing/e2e/gov/vest_test.go:35]: added more test cases like the ones mentioned here

File: protocol/testing/e2e/gov/vest_test.go [hunk 4]
```diff
@@ -106,149 +288,212 @@ func TestSetVestEntry(t *testing.T) {
...
+func TestDeleteVestEntry_Success(t *testing.T) {
+	tests := map[string]struct {
+		msgs                 []sdk.Msg
+		genesisVestEntryKeys []string // keys of vest entries in genesis state.
+	}{
+		"Success: delete one vest entry": {
+			msgs: []sdk.Msg{
+				&vesttypes.MsgDeleteVestEntry{
+					Authority:     authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					VesterAccount: "random_vester",
```
Comment (teddyding) [inline protocol/testing/e2e/gov/vest_test.go:330]: Nit: Can we define "random_vester" as a constant string (e.g. "TEST_VESTERT_ACCOUNT1 = "random_vester")

**Review by teddyding — APPROVED** [review]

**Pull request merged**